### PR TITLE
Stop setting `*Policies.remove=true` in `cilium-app`

### PR DESCRIPTION
**Pull request by kopiczko**
This pr: Stops setting `*Policies.remove=true` in `cilium-app`.

### Trigger e2e tests

/run cluster-test-suites

### CODE REVIEW

File: helm/cluster-vsphere/templates/helmreleases/cilium-helmrelease.yaml [hunk 1]
```diff
@@ -66,17 +66,6 @@ spec:
...
-      enabled: true
-      tolerations:
-        - effect: NoSchedule
-          operator: Exists
-        - effect: NoExecute
-          operator: Exists
-        - key: CriticalAddonsOnly
-          operator: Exists
-{{- end }}
     extraPolicies:
```
Comment (mcharriere) [inline helm/cluster-vsphere/templates/helmreleases/cilium-helmrelease.yaml:69]: probably would make sense to remove this block too

Comment (kopiczko) [reply helm/cluster-vsphere/templates/helmreleases/cilium-helmrelease.yaml:69]: Yeah :)

**Comment by kopiczko** [comment]
```
   ✖ [It] Basic upgrade test upgrade should apply new version successfully (601.658153791)
	Timed out after 600.001s.
	Expected
	    <bool>: false
	to be true
```

/run cluster-test-suites

**Review by vxav — CHANGES_REQUESTED** [review]
We need to keep the interface in the values in order not to break deployment of the netpol app
https://gigantic.slack.com/archives/C01BYMF6RN0/p1719213867953049?thread_ts=1718880813.764729&cid=C01BYMF6RN0

**Review by vxav — APPROVED** [review]

**Pull request merged**